File: Python-grpc/src/main/java/DataWarehouseClient.java
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

public class DataWarehouseClient {

    public static void main(String[] args) {

        ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", 50051)
                .usePlaintext()
                .build();

        DataWarehouseServiceGrpc.DataWarehouseServiceBlockingStub stub = DataWarehouseServiceGrpc.newBlockingStub(channel);

        DataWarehouse.WarehouseRequest dataRequest = DataWarehouse.WarehouseRequest.newBuilder()
                .setId(1).setAuth("IloveSYT".hashCode())
                .build();

        System.out.println("Request erfolgreich geschickt mit id: "+ dataRequest.getId());


        try {
            DataWarehouse.WarehouseResponse response = stub.warehousing(dataRequest);

            System.out.println("Response erhalten");
            System.out.println("Warehouse ID: " + response.getWarehouseID());
            System.out.println("Warehouse Name: " + response.getWarehouseName());
            System.out.println("Warehouse Storage: " + response.getWarehouseStorage());
            System.out.println("Warehouse Location: " + response.getWarehouseLocation());
            System.out.println("Warehouse Avaibility: " + response.getWarehouseAvailability());
            System.out.println("Warehouse Parkinslots: " + response.getWarehouseParkingSlots());
            response.getProductDataList().forEach(product -> {
                System.out.println("Product ID: " + product.getProductID() +
                        ", Name: " + product.getProductName() +
                        ", Category: " + product.getProductCategory() +
                        ", Price: " + product.getProductPrice() +
                        ", Stock: " + product.getProductStock() +
                        ", Expiry Date: " + product.getProductExpiryDate() +
                        ", Availability: " + product.getProductAvailability());
            });
        } catch (Exception e) {
            System.err.println("Fehler bei der Anfrage: " + e.getMessage());
        } finally {
            channel.shutdown();
        }
    }
}

File: Python-grpc/src/main/proto/DataWarehouseClient.py
import grpc
from DataWarehouse_pb2_grpc import DataWarehouseServiceStub
from DataWarehouse_pb2 import WarehouseRequest

def run():
    with grpc.insecure_channel('localhost:50051') as channel:
        stub = DataWarehouseServiceStub(channel)

        data_request = WarehouseRequest(
            id=1,
            auth=-1267811949
        )


        print(f"Request erfolgreich geschickt mit id: {data_request.id}")

        try:
            response = stub.warehousing(data_request)

            print("Response erhalten")
            print(f"Warehouse ID: {response.warehouseID}")
            print(f"Warehouse Name: {response.warehouseName}")
            print(f"Warehouse Storage: {response.warehouseStorage}")
            print(f"Warehouse Location: {response.warehouseLocation}")
            print(f"Warehouse Avaibility: {response.warehouseAvailability}")
            print(f"Warehouse Parkinslots: {response.warehouseParkingSlots}")

            for product in response.productData:
                print(f"Product ID: {product.productID}, "
                    f"Name: {product.productName}, "
                    f"Category: {product.productCategory}, "
                    f"Price: {product.productPrice}, "
                    f"Stock: {product.productStock}, "
                    f"Expiry Date: {product.productExpiryDate}, "
                    f"Availability: {product.productAvailability}")

        except grpc.RpcError as e:
            print(f"Fehler bei der Anfrage: {e.details()}")

if __name__ == '__main__':
    run()